packages/components/src/border-control/border-control/component.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * Internal dependencies
+ */
+import BorderControlDropdown from '../border-control-dropdown';
+import UnitControl from '../../unit-control';
+import RangeControl from '../../range-control';
+import { HStack } from '../../h-stack';
+import { StyledLabel } from '../../base-control/styles/base-control-styles';
+import { View } from '../../view';
+import { VisuallyHidden } from '../../visually-hidden';
+import { contextConnect, WordPressComponentProps } from '../../ui/context';
+import { useBorderControl } from './hook';
+
+import type { BorderControlProps, LabelProps } from '../types';
+
+const BorderLabel = ( props: LabelProps ) => {
+	const { label, hideLabelFromVision } = props;
+
+	if ( ! label ) {
+		return null;
+	}
+
+	return hideLabelFromVision ? (
+		<VisuallyHidden as="label">{ label }</VisuallyHidden>
+	) : (
+		<StyledLabel>{ label }</StyledLabel>
+	);
+};
+
+const BorderControl = (
+	props: WordPressComponentProps< BorderControlProps, 'div' >,
+	forwardedRef: React.ForwardedRef< any >
+) => {
+	const {
+		colors,
+		disableCustomColors,
+		enableAlpha,
+		enableStyle = true,
+		hideLabelFromVision,
+		innerWrapperClassName,
+		label,
+		onBorderChange,
+		onSliderChange,
+		onWidthChange,
+		placeholder,
+		previousStyleSelection,
+		sliderClassName,
+		value: border,
+		widthControlClassName,
+		widthUnit,
+		widthValue,
+		withSlider,
+		__experimentalHasMultipleOrigins,
+		__experimentalIsRenderedInSidebar,
+		...otherProps
+	} = useBorderControl( props );
+
+	return (
+		<View { ...otherProps } ref={ forwardedRef }>
+			<BorderLabel
+				label={ label }
+				hideLabelFromVision={ hideLabelFromVision }
+			/>
+			<HStack spacing={ 3 }>
+				<HStack className={ innerWrapperClassName }>
+					<BorderControlDropdown
+						border={ border }
+						colors={ colors }
+						disableCustomColors={ disableCustomColors }
+						enableAlpha={ enableAlpha }
+						enableStyle={ enableStyle }
+						onChange={ onBorderChange }
+						previousStyleSelection={ previousStyleSelection }
+						__experimentalHasMultipleOrigins={
+							__experimentalHasMultipleOrigins
+						}
+						__experimentalIsRenderedInSidebar={
+							__experimentalIsRenderedInSidebar
+						}
+					/>
+					<UnitControl
+						className={ widthControlClassName }
```

Comment:
Thanks the nudge on this. I tried adding `size="__unstable-large"` however it caused more issues than it solved. Instead I've just [set the `HStack`'s alignment to `stretch` ](https://github.com/WordPress/gutenberg/pull/37769/commits/202800c6b4f7739c99a8608a3366d848974b4060) which appears to achieve the desired visual result for me. 

Can you see any issues with this approach?

Reply:
Ah sorry, I thought we were going for the 40px size. We'll be introducing a new 36px default variant for `UnitControl` soon (#39397), which will be a fit for this case. Until then, `stretch` looks like a good solution 👍 

However, the paddings are going to be a problem, as they will increase from 8px to 16px by default. @jasmussen Should we change the specs on this `BorderControl` component so it has enough room for the 16px paddings? I'm assuming the wider paddings are also for providing larger touch targets in cases like this, where many interactive elements are close together.

Reply:
The closest matching design I found in [the Inspector Panels Figma file](https://www.figma.com/file/vmHxwa8WhiyiP9poe2om9J/Inspector-Panels?node-id=12571%3A26754) is the border radius, which works fine with 16px padding left and right:

<img width="292" alt="Screenshot 2022-03-16 at 15 24 47" src="https://user-images.githubusercontent.com/1204802/158612208-caf138a2-685e-4cfb-93fe-f6efffbdebcc.png">

I think a larger issue is that the circle inside the box doesn't match the metrics of the ItemGroup as designed in [Global Styles](https://www.figma.com/file/oEkcAyhIvPFMVEAO8EImvA/Global-Styles?node-id=1700%3A0):
<img width="372" alt="Screenshot 2022-03-16 at 15 26 05" src="https://user-images.githubusercontent.com/1204802/158612508-aaba5a5e-b231-4fcd-8b97-de3290687402.png">

That might be fine?

Reply:
> However, the paddings are going to be a problem, as they will increase from 8px to 16px by default.

This will be a bigger problem for the [`BorderBoxControl`](https://github.com/WordPress/gutenberg/pull/38876) where we have limited space when the individual border sides are "unlinked".

> Should we change the specs on this BorderControl component so it has enough room for the 16px padding?

All I've had to work from so far has been the images in https://github.com/WordPress/gutenberg/issues/35602. Anything more specific would definitely be a help and appreciated.

I also don't have access to the Global Styles figma link above.

Reply:
> I also don't have access to the Global Styles figma link above.

The link should have view access, but feel free to ping me in chat if you have trouble. In general I'll try to share screenshots here of the pertinent parts, though!

> This will be a bigger problem for the https://github.com/WordPress/gutenberg/pull/38876 where we have limited space when the individual border sides are "unlinked".

Definitely, seems fine for the near term to reduce those paddings. Longer term we [in the same Figma](https://www.figma.com/file/vmHxwa8WhiyiP9poe2om9J/Inspector-Panels?node-id=12103%3A16133) a couple of illustrations of how this could work:
<img width="1012" alt="Spacing" src="https://user-images.githubusercontent.com/1204802/158961413-08ddb22b-93e4-432b-8fc9-509aedb68697.png">

<img width="343" alt="Screenshot 2022-03-18 at 09 07 24" src="https://user-images.githubusercontent.com/1204802/158961427-dbd68ef4-9ff7-41ed-b14e-7cf371d3ebcf.png">

We might also want to keep it as a single row if possible, that's TBD, so again, fine to zero the paddings out in that state.

Reply:
The main question on my end is whether we want to make these reduced paddings an official part of `InputControl`'s API. If the reduced padding pattern is not something we want to encourage, I would probably recommend doing it with a style override here until the layout is reworked to accommodate the standard paddings.

Reply:
I would love for us to not make the override "official" quite yet, less is more in terms of the API.